File: leetcode/3798.cpp
class Solution {
  public:
    string largestEven(string s) {
        if (s.empty())
            return "";

        if (s.back() == '2')
            return std::string(s);

        return largestEven(s.substr(0, s.size() - 1));
    }
};

File: leetcode/3798.rs
impl Solution {
    pub fn largest_even(s: String) -> String {
        let n = s.len();

        if s == "".to_string() {
            return "".to_string();
        }

        if s.chars().last() == Some('2') {
            return s;
        }

        return Self::largest_even(s[..n-1].to_string());
    }
}
